### ISSUE TITLE
Toxic Exposure 1.0: Fix Validation issues found with Form526ToLighthouseTransform service

### ISSUE THREAD

**Issue by sethdarragile6** [department-of-veterans-affairs/va.gov-team#86252]
## Issue Description
Thomas and Dewayne of Carbs🥖 ran audits on the Lighthouse submit endpoint and reported their findings in the comments of [this ticket](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/84199) (which was originally created for 1 issue with treatment dates- which has already been resolved). This ticket is to fix these issues:

Issues found in Production (in priority Order):
 
 2. Response Status Code '422' - The property /directDeposit/accountNumber did not match the following requirements: 
 ```
 {"description"=>"Account number for the direct deposit.", "pattern"=>"^(?:[a-zA-Z0-9]{4,17})?$", "type"=>["string", "null"], "maxLength"=>14, "nullable"=>true, "example"=>"123123123123"}
 ```
**_Note: this may be the same as (14) from this ticket: #86496_**

 3. Response Status Code '422' - `Each treatment begin date must be after the first activeDutyBeginDate.`

 4. Response Status Code '422' - The property /veteranIdentification/mailingAddress/state did not match the following requirements: 
 ```
 {"description"=>"State for the Veteran's current mailing address.", "type"=>"string", "pattern"=>"^[a-z,A-Z]{2}$", "example"=>"OR"} 
 ```

 5. Response Status Code '422' - The property /veteranIdentification/mailingAddress/addressLine1 did not match the following requirements: 
 ```
 {"description"=>"Address line 1 for the Veteran's current mailing address.", "type"=>"string", "pattern"=>"^([-a-zA-Z0-9'.']([-a-zA-Z0-9'.' ])?)+$", "maxLength"=>20, "example"=>"1234 Couch Street"} 
 ```

 6. Response Status Code '422' - The property /treatments/0/center/city did not match the following requirements: 
 ```
 {"description"=>"City of treatment facility.", "type"=>["string", "null"], "pattern"=>"^$|^([-a-zA-Z'.' ])+$", "example"=>"Portland", "nullable"=>true} 
 ```

 8. Response Status Code '422' - The property /disabilities/0/name did not match the following requirements: 
 ```
 {"description"=>"Name of the disability. The /disabilities endpoint on the [Benefits Reference Data API](https://developer.va.gov/explore/benefits/docs/benefits_reference_data?version=current) may be used to retrieve possible disability names.", "type"=>"string", "pattern"=>"^$|([a-zA-Z0-9\"\\/&\\(\\)\\-'.,# ]([a-zA-Z0-9(\\)\\-'.,# ])?)+$", "example"=>"PTSD (post traumatic stress disorder)", "maxLength"=>255} 
 ```

 9. Response Status Code '422' - The property /directDeposit/financialInstitutionName did not match the following requirements: 
 ```
 {"description"=>"Provide the name of the financial institution where the Veteran wants the direct deposit.", "maxLength"=>35, "type"=>["string", "null"], "nullable"=>true, "example"=>"Some Bank"}
 ```

 10. Response Status Code '422' - The property /serviceInformation/servicePeriods did not match the following requirements: 
 ```
 {"description"=>"Identifies the Service dates and Branch the Veteran served in.", "type"=>"array", "minItems"=>1, "uniqueItems"=>true, "items"=>{"type"=>"object", "required"=>["serviceBranch", "serviceComponent", "activeDutyBeginDate", "activeDutyEndDate"], "properties"=>{"serviceBranch"=>{"description"=>"Branch of service during period. The /service-branches endpoint on the [Benefits Reference Data API](https://developer.va.gov/explore/benefits/docs/benefits_reference_data?version=current) may be used to retrieve list of possible service branches.", "type"=>"string", "maxLength"=>1000, "example"=>"Air Force"}, "serviceComponent"=>{"type"=>"string", "description"=>"", "enum"=>["Active", "Reserves", "National Guard"]}, "activeDutyBeginDate"=>{"description"=>"Date started active duty.", "type"=>"string", "pattern"=>"^(?:[0-9]{4})-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])$", "example"=>"2018-06-06"}, "activeDutyEndDate"=>{"description"=>"Date completed active duty.  If in the future, 'separationLocationCode' is required.", "type"=>"string", "pattern"=>"^(?:[0-9]{4})-(?:0[1-9]|1[0-2])-(?:0[1-9]|[1-2][0-9]|3[0-1])$", "example"=>"2018-06-06"}, "separationLocationCode"=>{"description"=>"Location code for the facility the Veteran plans to separate from. Required if 'servicePeriod.activeDutyEndDate' is in the future. Code must match the values returned by the /intake-sites endpoint on the [Benefits reference Data API](https://developer.va.gov/explore/benefits/docs/benefits_reference_data?version=current).", "type"=>["string", "null"], "nullable"=>true, "example"=>"98283"}}}} 
 ```

_Note: Issues found in [Staging and #1 and #7 moved here](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/86496)_

---
## Tasks
- [ ] Understand the outcomes of each of the above errors
- [ ] Determine which are launch blocking, which are not
- [ ] Split the non-launch blocking work into a separate ticket

## Acceptance Criteria
- [ ] Identified issues categorized/fixed/resolved.
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by alatempa24** [comment]
[Blocked by staging issue](https://dsva.slack.com/archives/C02CQP3RFFX/p1719338387498809)

**Comment by aurora-a-k-a-lightning** [comment]
**LH issues summary:**

1. Mailing Address1 Formatting
-- allow pound signs and periods in address line 1?
2. Mailing Address1 formatting - space at the end
-- LH validation fails if there is a space at the end of the address (i think?)
-- we can trim on our side...but...
3. Mailing Address - International Addresses
-- null state field values
-- zipFirstFive field has less than 5 
---- (could ITL addresses have more than 5 as well?)
4. Treatments and Active Duty
-- even though the activeDutyBeginDate is before the treatment date(s), this is failing validation
5. Treatments Center Name
-- this string did not pass validation: "Olin E. Teague Veterans' Center"
---- not sure why. i think it passes in Regex101
------ ^$|^([-a-zA-Z'.' ])+$
6. Direct Deposit Account Number
-- there are a lot of account numbers that are more than 14 characters
7. Direct Deposit Account Number Formatting
-- one submission had a dash "-" in the value
8. Direct Deposit Financial Institution Name
-- one submission had 36 characters where LH expects 35 characters or less

**Definitely us:**
9. Service Periods Duplication
-- story here
10. Mailing Address1 formatting - space at the end
-- LH validation fails if there is a space at the end of the address (i think?)
---- we can trim on our side...but...
------ story here

**Blurry lines:**
11. Disabilities Name
-- 526 allows users to type in disabilities, but LH validates against the BRD
---- feature mismatch?
